### PR TITLE
Clean up code and upgrade for Django 1.5.12 compatibility

### PR DESCRIPTION
**Pull request by ryanmrubin**
To be merged in unison with https://github.com/Crossway/federation/pull/107/.

### CODE REVIEW

**Comment by eykd** [comment]
👍

File: cas_provider/tests.py [hunk 1]
```diff
@@ -237,17 +226,18 @@ def test_cas2_custom_attrs(self):
...
                                            '''</cas:authenticationSuccess>'''
                                            '''</cas:serviceResponse>''')
 
 
     def test_cas2_fail_validate(self):
-        for user, pwd in (('root', '321'), ('notroot', '123'), ('nonactive', '123')):
-            response = self._login_user(user, pwd)
+        for user, pwd, email in (('root', '321', 'root@example.com'),
+                                 ('notroot', '123', 'notroot@example.com'),
+                                 ('nonactive', '123', 'nonactive@example.com')):
```
Comment (eykd) [inline cas_provider/tests.py]: Please move the tuple out to a variable. The for loop statement should never extend to multiple lines.

**Comment by adamzap** [comment]
👍

**Comment by dwickwire** [comment]
👍

**Comment by eykd** [comment]
👍

**Pull request merged**